crates/core/src/executor/mod.rs

```diff
@@ -129,7 +142,7 @@ impl<CodeDb: KVDatabase, ZkDb: KVDatabase + 'static> EvmExecutor<'_, '_, CodeDb,
             }
             env.tx.scroll.is_l1_msg = tx.is_l1_msg();
             let rlp_bytes = tx.rlp();
-            self.hooks.tx_rlp(self, &rlp_bytes);
+            tx_rlps.push(rlp_bytes.clone());
```

Comment:
Why do we need to clone here?

Reply:
We need to calculate l1 fee which is related to rlp bytes, also in chunk mode, we need calc the keccak of all tx rlp bytes. So it's for reusing the rlp bytes, avoid encoding it twice.